### ISSUE TITLE
Consider standard prefixes for domain objects

### ISSUE THREAD

**Issue by rwinch** [spring-projects/spring-security#4702]
### Summary

There are lots of very generic names for objects in OAuth. For example, `Address` is likely to be a domain object used in many applications within many different package scopes. The package name does help to clarify what it is, but this likely will collide with another Address object in an application which forces qualified class names.

It would likely be beneficial to come up with a standard prefix for all domain objects within each package. For example, OIDC domain objects can be prefixed with Oidc and OAuth2 objects can be prefixed with OAuth2

**Comment by rwinch** [comment]
@jgrandja Has this one been superseded by other issues?

**Comment by jgrandja** [comment]
@rwinch Sorry I missed this issue as it wasn't labeled.

This has been resolved via #4720 and #4722 as per our discussion.

I did rename all classes/interfaces we discussed and agreed upon with the exception of the following:

We agreed on `StandardClaimMemberNames` but I renamed to `StandardClaimNames`
We agreed on `IdTokenClaimMemberNames` but I renamed to `IdTokenClaimNames`

I just felt `Member` was redundant in the name and didn't read well. Let me know your thoughts on this and if you feel feel we should have `Member` in the name as discussed.
